packages/app/src/state/staking/actions.ts

```diff
@@ -351,7 +466,81 @@ export const unstakeKwenta = createAsyncThunk<void, BigNumber, ThunkConfig>(
 			txHash: hash,
 			onTxConfirmed: () => {
 				dispatch({ type: 'staking/setUnstakeStatus', payload: FetchStatus.Success })
-				dispatch(fetchStakingData())
+				dispatch(fetchStakeMigrateData())
+			},
+			onTxFailed: () => {
+				dispatch({ type: 'staking/setUnstakeStatus', payload: FetchStatus.Error })
+			},
+		})
+	}
+)
+
+export const stakeEscrowV2 = createAsyncThunk<void, BigNumber, ThunkConfig>(
+	'staking/stakeEscrow',
+	async (amount, { dispatch, extra: { sdk } }) => {
+		const { hash } = await sdk.kwentaToken.stakeEscrowedKwentaV2(amount)
+
+		monitorTransaction({
+			txHash: hash,
+			onTxConfirmed: () => {
+				dispatch({ type: 'staking/setStakeEscrowedStatus', payload: FetchStatus.Success })
+				dispatch(fetchStakeMigrateData())
+			},
+			onTxFailed: () => {
+				dispatch({ type: 'staking/setStakeEscrowedStatus', payload: FetchStatus.Error })
+			},
+		})
+	}
+)
+
+export const unstakeEscrowV2 = createAsyncThunk<void, BigNumber, ThunkConfig>(
+	'staking/unstakeEscrow',
+	async (amount, { dispatch, extra: { sdk } }) => {
+		const { hash } = await sdk.kwentaToken.unstakeEscrowedKwentaV2(amount)
+
+		monitorTransaction({
+			txHash: hash,
+			onTxConfirmed: () => {
+				dispatch({ type: 'staking/setUnstakeEscrowedStatus', payload: FetchStatus.Success })
+				dispatch(fetchStakeMigrateData())
+			},
+			onTxFailed: () => {
+				dispatch({ type: 'staking/setUnstakeEscrowedStatus', payload: FetchStatus.Error })
+			},
+		})
+	}
+)
+
+// TODO: Consider merging this with the (stake|unstake)Escrow actions.
+
+export const stakeKwentaV2 = createAsyncThunk<void, BigNumber, ThunkConfig>(
+	'staking/stakeKwenta',
+	async (amount, { dispatch, extra: { sdk } }) => {
+		const { hash } = await sdk.kwentaToken.stakeKwentaV2(amount)
+
+		monitorTransaction({
```

Comment:
Would be good if we could use a similar pattern to `monitorAndAwaitTransaction` so we are consistent with transaction handling throughout the app but something we can discuss and look at in a separate PR.

Reply:
Yes. That's a good idea. I will follow the pattern used in futures transactions.